test/Otoken.test.ts

```diff
@@ -0,0 +1,85 @@
+  let expiry: number
+  let expiryStringRedable: string
+  const isPut = true
+
+  before('Deployment', async () => {
+    // Need another mock contract for addressbook when we add ERC20 operations.
+    otoken = await Otoken.new(mockAddressBook)
+    usdc = await MockERC20.new('USDC', 'USDC')
+    expiry = (await time.latest()).add(time.duration.days(30)).toNumber()
+    expiryStringRedable = new Date(expiry * 1000)
+      .toISOString()
+      .split('T')[0]
+      .replace('-', '')
+      .replace('-', '')
+  })
+
+  describe('Otoken Initialization', () => {
+    it('should be able to initialize with put parameter', async () => {
+      await otoken.init(ETH_ADDR, usdc.address, usdc.address, strikePrice, expiry, isPut, {from: deployer})
+    })
+
+    it('should initilized the put option with valid name / symbol', async () => {
+      const name = await otoken.name()
+      const symbol = await otoken.symbol()
+      assert.equal(name, `ETHUSDC $200Put ${expiryStringRedable}`)
+      assert.equal(symbol, `ETHUSDC/${expiryStringRedable}/200P/USDC`)
+    })
+
+    it('should set the right name for calls', async () => {
+      const call = await Otoken.new(mockAddressBook)
+      await call.init(ETH_ADDR, usdc.address, usdc.address, strikePrice, expiry, false, {from: deployer})
+      const name = await call.name()
+      const symbol = await call.symbol()
+      assert.equal(name, `ETHUSDC $200Call ${expiryStringRedable}`)
+      assert.equal(symbol, `ETHUSDC/${expiryStringRedable}/200C/USDC`)
+    })
+
+    it('should display strikePrice as $0 in name and symbol when strikePrice < 10^18', async () => {
+      const testOtoken = await Otoken.new(mockAddressBook)
+      await testOtoken.init(ETH_ADDR, usdc.address, usdc.address, 0, expiry, isPut, {from: deployer})
+      const name = await testOtoken.name()
+      const symbol = await testOtoken.symbol()
+      assert.equal(name, `ETHUSDC $0Put ${expiryStringRedable}`)
+      assert.equal(symbol, `ETHUSDC/${expiryStringRedable}/0P/USDC`)
+    })
+
+    it('should get the correct parameters.', async () => {
+      const _underlying = await otoken.underlyingAsset()
+      const _strike = await otoken.strikeAsset()
+      const _collateral = await otoken.collateralAsset()
+      const _strikePrice = (await otoken.strikePrice()).toString()
+      const _isPut = await otoken.isPut()
+      const _expiry = (await otoken.expiry()).toNumber()
+      assert.equal(_underlying, ETH_ADDR)
+      assert.equal(_strike, usdc.address)
+      assert.equal(_collateral, usdc.address)
+      assert.equal(_strikePrice, strikePrice.toString())
+      assert.equal(_isPut, isPut)
+      assert.equal(_expiry, expiry)
+    })
+
+    it('should revert when init is called twice', async () => {
+      await expectRevert(
+        otoken.init(ETH_ADDR, usdc.address, usdc.address, strikePrice, expiry, isPut),
+        'Contract instance has already been initialized.',
```

Comment:
should there be a test on the upper limit of the date?

Reply:
Also is ETH going to be represented in the system as 0x0 or WETH? @haythem96 thoughts?

Reply:
We can use WETH and wrap from ETH to WETH using https://github.com/gnosis/canonical-weth, or we can think of another hack for the address(0) thing... let's decide this next eng sync!